packages/pigeon/lib/java_generator.dart

```diff
@@ -184,12 +186,20 @@ static MessageCodec<Object> getCodec() {
                 if (method.isAsynchronous) {
                   final String resultValue =
                       method.returnType.isVoid ? 'null' : 'result';
-                  methodArgument.add(
-                    'result -> { '
-                    'wrapped.put("${Keys.result}", $resultValue); '
-                    'reply.reply(wrapped); '
-                    '}',
-                  );
+                  const String resultName = 'resultCallback';
+                  indent.format('''
+Result<$returnType> $resultName = new Result<$returnType>() {
+\tpublic void success($returnType result) {
```

Comment:
This is a string argument of the Indent.format, all the tabs get changed into spaces when the code is generated.  Tabs were used in the string literal since the Indent class can adjust how many spaces and indent is.  It's one of the main jobs of Indent to abstract away the concept of how to indent code.